### PR TITLE
Add fallback map table

### PR DESCRIPTION
**Pull request by mck1117**
This change adds a table that estimates MAP based on RPM and TPS.  This isn't only useful for handling a failed MAP sensor, but could make acceleration enrichment MUCH easier to configure.

implement #932
related to #2244

### CODE REVIEW

File: firmware/controllers/algo/airmass/speed_density_airmass.cpp [hunk 1]
```diff
@@ -41,3 +41,21 @@ AirmassResult SpeedDensityAirmass::getAirmass(int rpm) {
 		map,	// AFR/VE table Y axis
 	};
 }
+
+float SpeedDensityAirmass::getMap(int rpm) const {
+	float fallbackMap;
+	if (CONFIG(enableMapEstimationTableFallback)) {
+		// if the map estimation table is enabled, estimate map based on the TPS and RPM
+		fallbackMap = m_mapEstimationTable->getValue(rpm, 100 * Sensor::get(SensorType::Tps1).value_or(0));
```
Comment (rusefillc) [inline firmware/controllers/algo/airmass/speed_density_airmass.cpp:49]: "100" magic constant, can we please give it a name or introduce a helpful macro?

**Comment by rusefillc** [comment]
@mck1117 poke "100" magic constant

File: firmware/controllers/algo/airmass/speed_density_airmass.cpp (hunk 1 above)
Comment (mck1117) [reply firmware/controllers/algo/airmass/speed_density_airmass.cpp]: magic is no longer magic :heavy_check_mark:

**Comment by rusefillc** [comment]
@mck1117 do we have default values for the new table?

**Comment by mck1117** [comment]
it's pretty engine specific - easiest to just log some normal driving and let megalogviewer generate the table for you

**Pull request merged**